### PR TITLE
build: Update NEWS in preperation for 3.1.0

### PR DESCRIPTION
**Pull request by bwbarrett**
Signed-off-by: Brian Barrett <bbarrett@amazon.com>

### CODE REVIEW

**Comment by bwbarrett** [comment]
This isn't perfect; we need to sync master against the release branches and then pull that back into the release branches (so this reflects 3.0.0 and possibly 3.0.1), but this should be the list that's new for 3.1.0.

File: NEWS [hunk 1]
```diff
@@ -53,9 +53,29 @@ included in the vX.Y.Z section and be denoted as:
    (** also appeared: A.B.C)  -- indicating that this item was previously
                                  included in release version vA.B.C.
 
-Master (not on release branches yet)
-------------------------------------
-
+3.1.0 -- January, 2017
```
Comment (jsquyres) [inline NEWS]: Man, I read this too quickly this morning: shouldn't that be 2018?

File: NEWS [hunk 2]
```diff
@@ -53,9 +53,29 @@ included in the vX.Y.Z section and be denoted as:
...
+----------------------
+
+- Add monitoring component for PML, OSC, and IO to track data movement
+  of MPI applications.
+- Add support for communicator assertions: mpi_assert_no_any_tag,
+  mpi_assert_no_any_source, mpi_assert_exact_length, and
+  mpi_assert_allow_overtaking.
+- Update PMIx to version 2.1.0.
+- Update hwloc to 1.11.6.
```
Comment (jsquyres) [inline NEWS]: Per Peter K's comments, this should be 1.11.7.

File: NEWS (hunk 1 above)
Comment (bwbarrett) [reply NEWS]: I figured inventing time travel was easier than getting an OMPI release out.  Didn't I tell you that's why I've been so behind in pull requests? :)

Yeah, fixed.

**Comment by bwbarrett** [comment]
@jsquyres can you re-review?  Thanks!

**Pull request merged**